examples/temperature-measurement-app/esp32/main/EchoServer.cpp

```diff
@@ -0,0 +1,260 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "esp_event.h"
+#include "esp_log.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "nvs_flash.h"
+
+#include <algorithm>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/param.h>
+
+#include "lwip/err.h"
+#include "lwip/sockets.h"
+#include "lwip/sys.h"
+#include <lwip/netdb.h>
+
+#include <inet/IPAddress.h>
+#include <inet/InetError.h>
+#include <inet/InetLayer.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/raw/Tuple.h>
+#include <transport/raw/UDP.h>
+
+#include "DataModelHandler.h"
+
+static const char * TAG = "echo_server";
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::Transport;
+
+extern const NodeId kLocalNodeId = 12344321;
+
+namespace {
+
+/**
+ * @brief implements something like "od -c", changes an arbitrary byte string
+ *   into a single-line of ascii.  Destroys any byte-wise encoding that
+ *   might be present, e.g. utf-8.
+ *
+ * @param bytes     potentially unprintable buffer
+ * @param bytes_len length of bytes
+ * @param out       where to put the printable string
+ * @param out_len   length of out
+ * @return size_t required size of output buffer, including null-termination
+ */
+static size_t odc(const uint8_t * bytes, size_t bytes_len, char * out, size_t out_len)
```

Comment:
This isn't needed, because we don't need the "echo" functionality at all, right?

Reply:
We are still passing information back on message received